packages/browser/src/client/orchestrator.ts

```diff
@@ -50,9 +50,14 @@ function createIframe(container: HTMLDivElement, file: string) {
   return iframe
 }
 
-async function done() {
-  await rpcDone()
-  await client.rpc.finishBrowserTests()
+async function done(testFinished = true) {
+  try {
+    await rpcDone()
+    await client.rpc.finishBrowserTests()
+  }
+  finally {
```

Comment:
Why is this needed? Finishing should be done in `onFinished` hook in UI

Reply:
not being called

Reply:
Maybe being called without files and with errors, I'm going to check what's happening, maybe the logic must be updated for browser mode.

Reply:
Calling `finish` resolves this promise:

https://github.com/vitest-dev/vitest/blob/07876b7e0f4f69cd071f0c89107ebdd08927e182/packages/vitest/src/node/pools/browser.ts#L59

Which in turn resolves this:

https://github.com/vitest-dev/vitest/blob/07876b7e0f4f69cd071f0c89107ebdd08927e182/packages/vitest/src/node/core.ts#L636

Which should always call `onFinished` here:

https://github.com/vitest-dev/vitest/blob/07876b7e0f4f69cd071f0c89107ebdd08927e182/packages/vitest/src/node/core.ts#L655

Reply:
`done` being called 3 times (I added a breakpoint)... will `onFinished` being called only once, the first time it is resolved? Then the client can receive some `onTaskUpdate` and the state will be reset to `running` again

Reply:
`onFinished` being called only once but we're receiving `onTaskUpdate` after `onFinished` called:

![imagen](https://github.com/vitest-dev/vitest/assets/6311119/30541502-39b4-435c-8905-dddd03bd87ad)

Reply:
no idea what's happening but now works again without those changes: I'm going to revert changes

Reply:
Looks like it is a bug in Vitest runner - it should wait until all `onTaskUpdate` are resolved before finishing `runTests`